pkg/generate/code/late_initialize.go

```diff
@@ -0,0 +1,188 @@
+func lateInitializedFieldNames(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+) []string {
+	fieldNames := make([]string, 0)
+	fieldNameToConfig := cfg.ResourceFields(r.Names.Original)
+	if len(fieldNameToConfig) > 0 {
+		for fName, fConfig := range fieldNameToConfig {
+			if fConfig != nil && fConfig.LateInitialize != nil {
+				fieldNames = append(fieldNames, fName)
+			}
+		}
+	}
+	return fieldNames
+}
+
+// LateInitializeFromReadOne returns the gocode to set LateInitialization fields from the ReadOne output
+// Field path separated by '.' indicates members in a struct
+// Field path separated by '..' indicates member/key in a map
+// Note: Unlike Map, updating individual element of a list is not supported. LateInitializing complete list is supported.
+func LateInitializeFromReadOne (
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	sourceKoVarName string,
+	targetKoVarName string,
+// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	// Sample generator config:
+	// fields:
+	//      Name:
+	//        late_initialize: {}
+	//      ImageScanningConfiguration.ScanOnPush:
+	//        late_initialize:
+	//          delay_seconds: 5
+	//      map..subfield.x:
+	//        late_initialize:
+	//          delay_seconds: 5
+	//      another.map..lastfield:
+	//        late_initialize:
+	//          delay_seconds: 5
+	//      some.list:
+	//        late_initialize:
+	//          delay_seconds: 10
+	//
+	// Sample output:
+	//if observed.Spec.ImageScanningConfiguration != nil && koWithDefaults.Spec.ImageScanningConfiguration != nil {
+	//	if observed.Spec.ImageScanningConfiguration.ScanOnPush != nil && koWithDefaults.Spec.ImageScanningConfiguration.ScanOnPush == nil {
+	//		koWithDefaults.Spec.ImageScanningConfiguration.ScanOnPush = observed.Spec.ImageScanningConfiguration.ScanOnPush
+	//	}
+	//}
+	//if observed.Spec.Name != nil && koWithDefaults.Spec.Name == nil {
+	//	koWithDefaults.Spec.Name = observed.Spec.Name
+	//}
+	//if observed.Spec.another != nil && koWithDefaults.Spec.another != nil {
+	//	if observed.Spec.another.map != nil && koWithDefaults.Spec.another.map != nil {
+	//		if observed.Spec.another.map[lastfield] != nil && koWithDefaults.Spec.another.map[lastfield] == nil {
+	//		koWithDefaults.Spec.another.map[lastfield] = observed.Spec.another.map[lastfield]
+	//	}
+	//	}
+	//}
+	//if observed.Spec.map != nil && koWithDefaults.Spec.map != nil {
+	//	if observed.Spec.map[subfield] != nil && koWithDefaults.Spec.map[subfield] != nil {
+	//	if observed.Spec.map[subfield].x != nil && koWithDefaults.Spec.map[subfield].x == nil {
+	//	koWithDefaults.Spec.map[subfield].x = observed.Spec.map[subfield].x
+	//}
+	//}
+	//}
+	//if observed.Spec.some != nil && koWithDefaults.Spec.some != nil {
+	//	if observed.Spec.some.list != nil && koWithDefaults.Spec.some.list == nil {
+	//		koWithDefaults.Spec.some.list = observed.Spec.some.list
+	//	}
+	//}
+	out := ""
+	lateInitializedFieldNames := lateInitializedFieldNames(cfg, r)
```

Comment:
not a big deal but this block is overriding the function name `lateInitializedFieldNames` in the function scope. Which makes it impossible to re-call.

Reply:
good point. renamed the function.